client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/step-content.tsx

```diff
@@ -9,11 +13,47 @@ type StepContentProps = {
 	goToStep?: NavigationControls[ 'goToStep' ];
 };
 
-const StepContent = ( { siteSlug, submit, goNext, goToStep }: StepContentProps ) => (
-	<div className="launchpad__content">
-		<Sidebar siteSlug={ siteSlug } submit={ submit } goNext={ goNext } goToStep={ goToStep } />
-		<LaunchpadSitePreview siteSlug={ siteSlug } />
-	</div>
-);
+function sortByRegistrationDateCompareFunction(
+	firstDomainObject: ResponseDomain,
+	secondDomainObject: ResponseDomain
+) {
+	if ( firstDomainObject.registrationDate > secondDomainObject.registrationDate ) {
+		return -1;
+	} else if ( firstDomainObject.registrationDate < secondDomainObject.registrationDate ) {
+		return 1;
+	}
+	return 1;
+}
+
+const StepContent = ( { submit, goNext, goToStep }: StepContentProps ) => {
+	const site = useSite();
+
+	const { data: allDomains = [] } = useGetDomainsQuery( site?.ID ?? null, {
+		retry: false,
+	} );
+
+	const domains = allDomains.map( createSiteDomainObject );
+
+	// Filter custom domains and sort by registration date.
+	// We want the first domain object to be the most recently registered domain
+	const nonWpcomDomains = domains
+		.filter( ( domain ) => ! domain.isWPCOMDomain )
+		.sort( sortByRegistrationDateCompareFunction );
+
+	const wpcomDomains = domains.filter( ( domain ) => domain.isWPCOMDomain );
+
+	const sidebarURL = nonWpcomDomains?.length
+		? nonWpcomDomains[ 0 ]?.domain
+		: wpcomDomains[ 0 ]?.domain;
+
+	const iFrameURL = wpcomDomains.length ? wpcomDomains[ 0 ]?.domain : nonWpcomDomains[ 0 ]?.domain;
```

Comment:
From the way things are coded with the null propagation operators, it appears as if iframeURL and sidebarURL may return `undefined`? Should we have a graceful fallback for either case?

Reply:
Yeah, we were discussing that with Anthony but we thought that this would be more of a business/design decision because we need to know what will happen to the UI when there's no iFrame/sidebar domain... For example we might need to add a loading state for the sidebar domain container because it just stays empty while the domain is being fetched

Reply:
And since we are assuming both of those could not exist we need to know what will happen to the UI. But if this is such a unlikely scenario I guess we can just conditionally render the sidebar domain container and the iFrame preview to only render if there's a domain (which should be always the case)

Reply:
> we need to know what will happen to the UI when there's no iFrame/sidebar domain... For example we might need to add a loading state for the sidebar domain container because it just stays empty while the domain is being fetched

Wondering if @nuriapenya has thoughts

Reply:
We discussed this during our team meeting and decided it isn't worth looking into as much right now since it's not a likely situation if the user is going through the normal flow.